src/JsonLd/Serializer/ObjectNormalizer.php

```diff
@@ -84,7 +85,12 @@ public function normalize($object, $format = null, array $context = [])
         }
 
         if (isset($originalResource)) {
-            $context['output']['iri'] = $this->iriConverter->getIriFromItem($originalResource);
+            try {
+                $context['output']['iri'] = $this->iriConverter->getIriFromItem($originalResource);
+            } catch (InvalidArgumentException $e) {
+                $context['output']['iri'] = '_:'.(\function_exists('spl_object_id') ? spl_object_id($originalResource) : spl_object_hash($originalResource));
```

Comment:
I don't like exposing such internal PHP details. I would prefer generating a random it or, at least a hash of this value.